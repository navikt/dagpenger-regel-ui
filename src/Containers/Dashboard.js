import React, { useState, useEffect } from 'react';
import axios from 'axios';

const Dashboard = (
  location,
) => {
  const routeParams = location.match.params;
  const b = 'dsfdsfdsfd';
  const { aktorId, beregningdato, vilkarid } = routeParams;

  const [data, setData] = useState({ inntekter: {} });

  useEffect(async () => {
    const result = await axios(
      'http://localhost:3000/mock/inntekter.json',
    );

    setData(result.data);
  });

  return (
    <div />
  );
};

export default Dashboard;
