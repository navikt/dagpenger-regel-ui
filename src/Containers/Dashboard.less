@import '~nav-frontend-core/less/_variabler';

@width: 300px;
@height: 300px;

.grid {
  display: flex;
  flex-direction: row;
  justify-content: start;
  overflow-x: auto;
  width: 100%;

  .item {
    padding: 16px;
    border-right: 1px solid black;
    border-top: 1px solid black;
  }

  .arbeidsgivere {
    background: @navLysGra;
    width: @width;
    position: absolute;
    left: 0;

    .arbeidsgiver {
      height: @height;
    }
  }

  .maaneder {
    display: flex;
    flex-direction: row;
    justify-content: start;
    overflow-x: auto;
    margin-left: @width;
    width: 100%;

    .maaned {
      background: white;
      width: @width;

      .maanednavn {
        background: @navMorkGra;
        color: white;
      }

      .inntekter {
        height: @height;

        .inntekt {

        }
      }
    }
  }
}
