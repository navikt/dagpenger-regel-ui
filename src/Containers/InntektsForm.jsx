import React, { useState, useContext } from 'react';
import PropTypes from 'prop-types';
import { Form, withFormik, FieldArray } from 'formik';
import AlertStripe from 'nav-frontend-alertstriper';
import isEqual from 'lodash.isequal';
import { Element, Undertekst } from 'nav-frontend-typografi';
import Modal from 'nav-frontend-modal';
import { Hovedknapp, Knapp } from 'nav-frontend-knapper';
import Spacer from '../Components/Spacer';
import { Inntektstabell } from './Inntektstabell';
import { lagreInntekt } from '../lib/inntektApiClient';
import NyArbeidsgiver from './NyArbeidsgiver';
import OkAvbrytModal from '../Components/OkAvbrytModal';
import { LocaleContext } from '../Context/Locale';

const InntektsForm = (props) => {
  const [isArbeidsgiverModalOpen, setArbeidsgiverModal] = useState(false);
  const [isBekreftModalOpen, setBekreftModal] = useState(false);
  const locale = useContext(LocaleContext);

  const {
    hentInntektStatus, values, dirty, readOnly, handleSubmit, status, errors, isSubmitting,
  } = props;

  const { arbeidsgivere } = values;
  return (
    <>
      {status && status.success && (
      <div aria-live="polite">
        <AlertStripe type="suksess">
          <Element>{locale.inntektLagret}</Element>
          {locale.inntektLagretHuskÅBeregneIArena}
        </AlertStripe>
        <Spacer sixteenPx />
      </div>
      )}
      <Form onSubmit={handleSubmit}>
        <Inntektstabell
          {...props}
          readOnly={readOnly}
          hentInntektStatus={hentInntektStatus}
        />
        {errors.name && <div className="error">{errors.name}</div>}

        <div className="toolbar flex knapprad">
          <div className="leggtilarbeidsgiver">
            <Knapp
              htmlType="button"
              mini
              disabled={readOnly}
              onClick={() => setArbeidsgiverModal(!isArbeidsgiverModalOpen)}
            >
              Legg til arbeidsgiver
            </Knapp>
            <Modal
              isOpen={isArbeidsgiverModalOpen}
              onRequestClose={() => setArbeidsgiverModal(false)}
              closeButton={false}
              contentLabel="Legg til arbeidsgiver"
              ariaHideApp={false}
            >
              <FieldArray
                name="arbeidsgivere"
                render={arrayHelpers => (
                  <NyArbeidsgiver
                    arbeidsgivere={arbeidsgivere}
                    arrayHelpers={arrayHelpers}
                    closeModal={() => setArbeidsgiverModal(false)}
                  />
                )}
              />
            </Modal>

          </div>

          <div className="flexend flex">
            <div className="w200 marginhoyre16"><Undertekst>Du må bekrefte at de nye opplysningene skal benyttes.</Undertekst></div>
            {values.redigertAvSaksbehandler && (
            <Hovedknapp
              htmlType="button"
              onClick={() => setBekreftModal(true)}
              spinner={isSubmitting}
              autoDisableVedSpinner
              disabled={!hentInntektStatus && !dirty}
            >
              Bekreft
            </Hovedknapp>
            )}
            {!values.redigertAvSaksbehandler && (
            <Hovedknapp
              htmlType="submit"
              onClick={handleSubmit}
              spinner={isSubmitting}
              autoDisableVedSpinner
              disabled={!hentInntektStatus && !dirty}
            >
              Bekreft
            </Hovedknapp>
            )}
            <OkAvbrytModal
              isOpen={isBekreftModalOpen}
              text="Når du bekrefter ny inntekt så vil alle tidligere endringene overskrives."
              avbrytCallback={() => setBekreftModal(false)}
              OkCallback={() => {
                handleSubmit();
                setBekreftModal(false);
              }}
            />

          </div>
        </div>

      </Form>

    </>
  );
};

InntektsForm.propTypes = {
  readOnly: PropTypes.bool.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  isSubmitting: PropTypes.bool.isRequired,
  errors: PropTypes.shape().isRequired,
  locationData: PropTypes.shape().isRequired,
  dirty: PropTypes.bool.isRequired,
  values: PropTypes.shape().isRequired,
  hentInntektStatus: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.bool,
  ]).isRequired,
  status: PropTypes.shape(),
};

InntektsForm.defaultProps = {
  status: undefined,
};

export default withFormik({
  mapPropsToValues: (props) => {
    const {
      initialValues,
    } = props;
    return (initialValues);
  },


  handleSubmit: (values, formProps) => {
    const dirty = !isEqual(formProps.props.initialValues, values);

    setTimeout(() => {
      lagreInntekt(
        { ...values, redigertAvSaksbehandler: dirty || values.manueltRedigert },
        formProps.props.hentInntektStatus,
        formProps.props.locationData,
      )
        .then((result) => {
          // setInntektdata({ ...result.data });
          // setArbeidsgivere(findArbeidsgivere(result.data.inntekt));
          formProps.setStatus({ success: true });
          formProps.setSubmitting(false);
        })
        .catch((error) => {
          formProps.setStatus({ failure: true });
          formProps.setError(error);
        });
    }, 1000);
  },

  validate: true,
  displayName: 'InntektsForm',
  enableReinitialize: true,

})(InntektsForm);
