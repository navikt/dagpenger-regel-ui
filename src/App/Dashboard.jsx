import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useQuery } from '@apollo/react-hooks';
import Panel from 'nav-frontend-paneler';
import { Element, Undertekst, Undertittel } from 'nav-frontend-typografi';
import { loader } from 'graphql.macro';
import { ReactComponent as MannIkon } from '../images/mann.svg';
import { ReactComponent as KvinneIkon } from '../images/kvinne.svg';
import InntektsForm from '../Inntekt/InntektsForm';
import EditedIkon from '../Components/EditedIkon';
import Spinner from '../Components/Spinner';
import Spacer from '../Components/Spacer';
import { getAlleMåneder } from '../Utils/datoUtils';

const GET_INNTEKT = loader('../Graphql/GET_INNTEKT.gql');

const getKjønn = kjoenn => {
  if (kjoenn === 'Kvinne') {
    return <KvinneIkon />;
  }

  return <MannIkon />;
};

const inntektRequest = queryParams => ({
  personId: queryParams.get('aktorId'),
  vedtakId: queryParams.get('vedtakId'),
  beregningsdato: queryParams.get('beregningdato'),
});

// todo flytte til datoutils
export const set36Måneder = (data, beregningsmåneder) => {
  if (beregningsmåneder) {
    data.map(virksomhet => {
      if (virksomhet.posteringer === undefined) {
        Object.assign(virksomhet, {
          posteringer: {},
        });
      }

      beregningsmåneder.forEach(måned => {
        if (!virksomhet.posteringer[måned]) {
          Object.assign(virksomhet.posteringer, {
            [måned]: [],
          });
        }
      });
      return data;
    });
  }
  return data;
};

// TODO generaliser slik at den kan fortsatt være i utils
const groupBy = (list, keyGetter) => {
  const map = new Map();
  list.forEach(item => {
    const key = keyGetter(item);
    const collection = map.get(key);
    const dato = `${item.aarMaaned}`;
    if (!collection) {
      map.set(key, {
        ...item.virksomhet,
        posteringer: { [dato]: [item] },
      });
    } else if (collection.posteringer[dato]) {
      collection.posteringer[dato].push(item);
    } else {
      Object.assign(collection.posteringer, {
        [dato]: [item],
      });
    }
  });
  return Array.from(map.values());
};

const Dashboard = ({ readOnly, location }) => {
  const [hentInntektStatus, setHentInntekttatus] = useState(false);
  const [isHentInntektModalOpen, setHentInntektModal] = useState(false);

  const { data, error, loading, refetch } = useQuery(GET_INNTEKT, {
    variables: { ...inntektRequest(new URLSearchParams(location.search)), forceRefresh: hentInntektStatus },
  });

  const fetchUncachedInntekt = async () => {
    await setHentInntekttatus(true);
    // hent ferske inntekter
    refetch();
  };

  if (loading) {
    return <Spinner type="XL" />;
  }

  if (error || !data.person) {
    throw new Error(error);
  }

  const { person } = data;

  const beregningsmåneder = getAlleMåneder(person.vedtak.inntekt.fraDato, person.vedtak.inntekt.tilDato);

  const inntekter =
    !loading &&
    set36Måneder(
      groupBy(person.vedtak.inntekt.posteringer, postering => postering.virksomhet.organisasjonsnummer || postering.virksomhet.naturligIdent),
      beregningsmåneder,
    );

  return (
    <>
      <Panel border>
        <div className="flex">
          {person.naturligIdent && (
            <>
              <div className="marginhoyre16">{getKjønn(person.kjoenn)}</div>
              <div>
                <Undertittel>{`${person.navn} (${person.alder} år)`}</Undertittel>
                <Undertekst>{person.naturligIdent}</Undertekst>
              </div>
            </>
          )}
          <div className="flexend flex noprint">
            {person.vedtak.inntekt.manueltRedigert && (
              <div className="marginhoyre16 flex">
                <EditedIkon />
                <Element>Manuelt redigert</Element>
              </div>
            )}
          </div>
        </div>
      </Panel>

      <Spacer sixteenPx />

      <InntektsForm
        locationData={inntektRequest(new URLSearchParams(location.search))}
        initialValues={{
          person: {
            ...person,
            vedtak: {
              ...person.vedtak,
              inntekt: {
                ...person.vedtak.inntekt,
                posteringer: [...inntekter],
              },
            },
          },
        }}
        hentInntektStatus={hentInntektStatus}
        måneder={beregningsmåneder}
        readOnly={readOnly}
        isHentInntektModalOpen={isHentInntektModalOpen}
        setHentInntektModal={setHentInntektModal}
        fetchUncachedInntekt={fetchUncachedInntekt}
      />
    </>
  );
};

Dashboard.propTypes = {
  readOnly: PropTypes.bool,
  location: PropTypes.shape().isRequired,
};

Dashboard.defaultProps = {
  readOnly: false,
};

export default Dashboard;
