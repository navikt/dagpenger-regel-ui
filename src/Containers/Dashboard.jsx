import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { Formik } from 'formik';
import axios from 'axios';
import { Knapp } from 'nav-frontend-knapper';
import AlertStripe from 'nav-frontend-alertstriper';
import { Normaltekst, Ingress } from 'nav-frontend-typografi';
import Panel from 'nav-frontend-paneler';
import { ToggleKnapp } from 'nav-frontend-toggle';
import Spinner from '../Components/Spinner';
import Spacer from '../Components/Spacer';
import { Inntektstabell } from './Inntektstabell';
import { OkAvbrytModal } from '../Components/OkAvbrytModal';
import {
  getInntekt, getUncachedInntekt, lagreInntekt, getName,
} from '../lib/inntektApiClient';
import { ReactComponent as MannIkon } from '../images/mann.svg';

import './Dashboard.css';

export const findArbeidsgivere = (inntekt) => {
  const map = new Map();
  inntekt.arbeidsInntektMaaned
    .forEach(mnd => mnd.arbeidsInntektInformasjon.inntektListe
      .forEach((arbeidsgiver) => {
        // TODO fikse denne, bør flyttes til backend ved et senere tidspunkt
        const navn = 'test'; // getName(JSON.stringify({ id: arbeidsgiver.virksomhet.identifikator, aktørType: arbeidsgiver.virksomhet.aktoerType }));

        map.set(arbeidsgiver.virksomhet.identifikator, { navn, ...arbeidsgiver.virksomhet });
      }));

  return Array.from(map.values())
    .sort((a, b) => b.identifikator - a.identifikator);
};


const inntektRequest = queryParams => ({
  aktørId: queryParams.get('aktorId'),
  vedtakId: queryParams.get('vedtakId'),
  beregningsDato: queryParams.get('beregningdato'),
});

const Dashboard = ({ readOnly, location }) => {
  const [inntektdata, setInntektdata] = useState({ inntektId: '', inntekt: { arbeidsInntektMaaned: [], ident: {}, manueltRedigert: false } });
  const [arbeidsgivere, setArbeidsgivere] = useState([]);
  const [hentInntektStatus, setHentInntekttatus] = useState(false);
  const [isHentInntektModalOpen, setHentInntektModal] = useState(false);


  useEffect(() => {
    const getInntektFromApi = async () => {
      let result;
      if (process.env.NODE_ENV !== 'production') {
        result = await axios(
          `${process.env.PUBLIC_URL}/mock/mock.json`,
        );
      } else {
        result = await getInntekt(inntektRequest(new URLSearchParams(location.search)));
      }
      setInntektdata({ ...result.data });
      setArbeidsgivere(findArbeidsgivere(result.data.inntekt));
    };

    getInntektFromApi();
  }, [location.search]);

  const fetchUncachedInntekt = async () => {
    setHentInntekttatus('fetching');
    if (process.env.NODE_ENV !== 'production') {
      const result = await axios(
        `${process.env.PUBLIC_URL}/mock/mock1.json`,
      );
      setInntektdata({ ...result.data });
      setArbeidsgivere(findArbeidsgivere(result.data.inntekt));
      setHentInntekttatus(true);
    } else {
      getUncachedInntekt(inntektRequest(new URLSearchParams(location.search)))
        .then((result) => {
          setInntektdata({ ...result.data });
          setArbeidsgivere(findArbeidsgivere(result.data.inntekt));
          setHentInntekttatus(true);
        })
        // eslint-disable-next-line no-unused-vars
        .catch((error) => {
          setHentInntekttatus('error');
        });
    }
  };

  if (!arbeidsgivere.length || !inntektdata.inntekt.arbeidsInntektMaaned.length) {
    return <Spinner type="XL" />;
  }

  return (
    <>
      <Panel border>
        <div className="flex">
          <MannIkon />
          <div className="paddingleft">
            <Ingress>Ola Nordmann</Ingress>
            <Normaltekst>{`AktørId: ${inntektdata.inntekt.ident.identifikator}`}</Normaltekst>
          </div>
          <div className="flexend hoyre">
            InntektId
            <Normaltekst>{inntektdata.inntektId.id}</Normaltekst>
          </div>
        </div>
      </Panel>

      <Spacer sixteenPx />

      {hentInntektStatus
      && <div aria-live="polite"><AlertStripe type="info">Ny inntekt hentet</AlertStripe></div>}
      {hentInntektStatus === 'error'
      && (
        <div aria-live="polite">
          <AlertStripe type="feil">En feil skjedde under henting av frisk inntekt</AlertStripe>
        </div>
      )}

      <div className="flex">
        <Knapp
          onClick={() => setHentInntektModal(true)}
          autoDisableVedSpinner
          spinner={hentInntektStatus === 'fetching'}
        >

        Hent inntekter på nytt
        </Knapp>

        <OkAvbrytModal
          isOpen={isHentInntektModalOpen}
          text="Når du henter inn nyeste inntekt fra skatt så vil alle endringene dine gå tapt."
          avbrytCallback={() => setHentInntektModal(false)}
          OkCallback={() => {
            fetchUncachedInntekt();
            setHentInntektModal(false);
          }}
        />
        <div className="flexend">
          <ToggleKnapp>Vis alle</ToggleKnapp>
        </div>

      </div>


      <Spacer sixteenPx />

      <Formik
        validate
        enableReinitialize
        initialValues={{
          ...inntektdata,
          arbeidsgivere: [
            ...arbeidsgivere,
          ],
        }}
        onSubmit={(values, actions) => {
          setTimeout(() => {
            lagreInntekt(values)
              .then((result) => {
                setInntektdata({ ...result.data });
                setArbeidsgivere(findArbeidsgivere(result.data.inntekt));
                actions.setStatus({ success: true });
              })
              // eslint-disable-next-line no-unused-vars
              .catch((error) => {
                actions.setStatus({ failure: true });
              }).finally(() => {
                actions.setSubmitting(false);
              });
          }, 1000);
        }}
        render={formProps => (
          <Inntektstabell
            {...formProps}
            readOnly={readOnly}
            hentInntektStatus={hentInntektStatus}
          />
        )}
      />

    </>
  );
};

Dashboard.propTypes = {
  location: PropTypes.shape({ search: PropTypes.string.isRequired }).isRequired,
  readOnly: PropTypes.bool,
};

Dashboard.defaultProps = { readOnly: false };

export default Dashboard;
