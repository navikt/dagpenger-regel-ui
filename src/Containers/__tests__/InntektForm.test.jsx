import React from 'react';
import renderer from 'react-test-renderer';
import InntektsForm from '../InntektsForm';
import { findArbeidsgivere } from '../Dashboard';
import inntektMock from '../../../public/mock/mock';

const arbeidsgivere = findArbeidsgivere(inntektMock.inntekt);

const locationData = {
  aktÃ¸rId: '123',
  vedtakId: '123',
  beregningsDato: '2019-06',
};

describe('InntektsForm', () => {
  xtest('Skal vise ny InntektsForm', () => {
    const component = renderer.create(
      <InntektsForm
        readOnly={false}
        hentInntektStatus={false}
        inntektdata={inntektMock}
        locationData={locationData}
        initialValues={{
          ...inntektMock,
          arbeidsgivere: [
            ...arbeidsgivere,
          ],
        }}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
