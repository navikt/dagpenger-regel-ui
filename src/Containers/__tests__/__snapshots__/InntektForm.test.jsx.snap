// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InntektForm Skal vise ny InntektForm 1`] = `
<form
  onReset={[Function]}
  onSubmit={[Function]}
>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "
  .grid {
    grid-template-areas: \\". maaned--2017-07 maaned--2017-08 maaned--2017-09 maaned--2017-10 maaned--2017-11 maaned--2017-12 \\" \\"arbeidsgiver--222222222 inntekter--222222222--2017-07  inntekter--222222222--2017-08  inntekter--222222222--2017-09  inntekter--222222222--2017-10  inntekter--222222222--2017-11  inntekter--222222222--2017-12 \\" \\"arbeidsgiver--111111111 inntekter--111111111--2017-07  inntekter--111111111--2017-08  inntekter--111111111--2017-09  inntekter--111111111--2017-10  inntekter--111111111--2017-11  inntekter--111111111--2017-12 \\";
",
      }
    }
  />
  <div
    className="grid"
    id="grid"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
.arbeidsgiver--222222222 {
  grid-area: arbeidsgiver--222222222;
}
",
        }
      }
    />
    <div
      className="item arbeidsgiver arbeidsgiver--222222222"
    >
      <h2
        className="typo-undertittel"
      >
        NAVN
      </h2>
      <p
        className="typo-etikett-liten"
      >
        Org. nr
      </p>
      <p
        className="typo-element"
      >
        222222222
      </p>
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
.arbeidsgiver--111111111 {
  grid-area: arbeidsgiver--111111111;
}
",
        }
      }
    />
    <div
      className="item arbeidsgiver arbeidsgiver--111111111"
    >
      <h2
        className="typo-undertittel"
      >
        NAVN
      </h2>
      <p
        className="typo-etikett-liten"
      >
        Org. nr
      </p>
      <p
        className="typo-element"
      >
        111111111
      </p>
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
.maaned--2017-07 {
  grid-area: maaned--2017-07;
}
",
        }
      }
    />
    <div
      className="item maaned maaned--2017-07"
    >
      juli 2017
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
.inntekter--222222222--2017-07 {
  grid-area: inntekter--222222222--2017-07;
}
",
        }
      }
    />
    <div
      className="item inntekter inntekter--222222222--2017-07"
    >
      <div
        className="ekspanderbartPanel ekspanderbartPanel--lukket"
      >
        <button
          aria-expanded={false}
          className="ekspanderbartPanel__hode"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <div
            className="ekspanderbartPanel__flex-wrapper"
          >
            <span
              className="typo-ingress ekspanderbartPanel__heading"
            >
              NOK 6 410,26
            </span>
            <span
              className="ekspanderbartPanel__indikator"
            />
          </div>
        </button>
      </div>
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
.inntekter--111111111--2017-07 {
  grid-area: inntekter--111111111--2017-07;
}
",
        }
      }
    />
    <div
      className="item inntekter inntekter--111111111--2017-07"
    >
      <div
        className="ekspanderbartPanel ekspanderbartPanel--lukket"
      >
        <button
          aria-expanded={false}
          className="ekspanderbartPanel__hode"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <div
            className="ekspanderbartPanel__flex-wrapper"
          >
            <span
              className="typo-ingress ekspanderbartPanel__heading"
            >
              NOK 89 743,60
            </span>
            <span
              className="ekspanderbartPanel__indikator"
            />
          </div>
        </button>
      </div>
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
.maaned--2017-08 {
  grid-area: maaned--2017-08;
}
",
        }
      }
    />
    <div
      className="item maaned maaned--2017-08"
    >
      august 2017
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
.inntekter--222222222--2017-08 {
  grid-area: inntekter--222222222--2017-08;
}
",
        }
      }
    />
    <div
      className="item inntekter inntekter--222222222--2017-08"
    >
      <div
        className="ekspanderbartPanel ekspanderbartPanel--lukket"
      >
        <button
          aria-expanded={false}
          className="ekspanderbartPanel__hode"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <div
            className="ekspanderbartPanel__flex-wrapper"
          >
            <span
              className="typo-ingress ekspanderbartPanel__heading"
            >
              NOK 0,00
            </span>
            <span
              className="ekspanderbartPanel__indikator"
            />
          </div>
        </button>
      </div>
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
.inntekter--111111111--2017-08 {
  grid-area: inntekter--111111111--2017-08;
}
",
        }
      }
    />
    <div
      className="item inntekter inntekter--111111111--2017-08"
    >
      <div
        className="ekspanderbartPanel ekspanderbartPanel--lukket"
      >
        <button
          aria-expanded={false}
          className="ekspanderbartPanel__hode"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <div
            className="ekspanderbartPanel__flex-wrapper"
          >
            <span
              className="typo-ingress ekspanderbartPanel__heading"
            >
              NOK 0,00
            </span>
            <span
              className="ekspanderbartPanel__indikator"
            />
          </div>
        </button>
      </div>
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
.maaned--2017-09 {
  grid-area: maaned--2017-09;
}
",
        }
      }
    />
    <div
      className="item maaned maaned--2017-09"
    >
      september 2017
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
.inntekter--222222222--2017-09 {
  grid-area: inntekter--222222222--2017-09;
}
",
        }
      }
    />
    <div
      className="item inntekter inntekter--222222222--2017-09"
    >
      <div
        className="ekspanderbartPanel ekspanderbartPanel--lukket"
      >
        <button
          aria-expanded={false}
          className="ekspanderbartPanel__hode"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <div
            className="ekspanderbartPanel__flex-wrapper"
          >
            <span
              className="typo-ingress ekspanderbartPanel__heading"
            >
              NOK 54 194,06
            </span>
            <span
              className="ekspanderbartPanel__indikator"
            />
          </div>
        </button>
      </div>
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
.inntekter--111111111--2017-09 {
  grid-area: inntekter--111111111--2017-09;
}
",
        }
      }
    />
    <div
      className="item inntekter inntekter--111111111--2017-09"
    >
      <div
        className="ekspanderbartPanel ekspanderbartPanel--lukket"
      >
        <button
          aria-expanded={false}
          className="ekspanderbartPanel__hode"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <div
            className="ekspanderbartPanel__flex-wrapper"
          >
            <span
              className="typo-ingress ekspanderbartPanel__heading"
            >
              NOK 54 194,06
            </span>
            <span
              className="ekspanderbartPanel__indikator"
            />
          </div>
        </button>
      </div>
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
.maaned--2017-10 {
  grid-area: maaned--2017-10;
}
",
        }
      }
    />
    <div
      className="item maaned maaned--2017-10"
    >
      oktober 2017
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
.inntekter--222222222--2017-10 {
  grid-area: inntekter--222222222--2017-10;
}
",
        }
      }
    />
    <div
      className="item inntekter inntekter--222222222--2017-10"
    >
      <div
        className="ekspanderbartPanel ekspanderbartPanel--lukket"
      >
        <button
          aria-expanded={false}
          className="ekspanderbartPanel__hode"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <div
            className="ekspanderbartPanel__flex-wrapper"
          >
            <span
              className="typo-ingress ekspanderbartPanel__heading"
            >
              NOK 36 886,20
            </span>
            <span
              className="ekspanderbartPanel__indikator"
            />
          </div>
        </button>
      </div>
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
.inntekter--111111111--2017-10 {
  grid-area: inntekter--111111111--2017-10;
}
",
        }
      }
    />
    <div
      className="item inntekter inntekter--111111111--2017-10"
    >
      <div
        className="ekspanderbartPanel ekspanderbartPanel--lukket"
      >
        <button
          aria-expanded={false}
          className="ekspanderbartPanel__hode"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <div
            className="ekspanderbartPanel__flex-wrapper"
          >
            <span
              className="typo-ingress ekspanderbartPanel__heading"
            >
              NOK 36 886,20
            </span>
            <span
              className="ekspanderbartPanel__indikator"
            />
          </div>
        </button>
      </div>
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
.maaned--2017-11 {
  grid-area: maaned--2017-11;
}
",
        }
      }
    />
    <div
      className="item maaned maaned--2017-11"
    >
      november 2017
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
.inntekter--222222222--2017-11 {
  grid-area: inntekter--222222222--2017-11;
}
",
        }
      }
    />
    <div
      className="item inntekter inntekter--222222222--2017-11"
    >
      <div
        className="ekspanderbartPanel ekspanderbartPanel--lukket"
      >
        <button
          aria-expanded={false}
          className="ekspanderbartPanel__hode"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <div
            className="ekspanderbartPanel__flex-wrapper"
          >
            <span
              className="typo-ingress ekspanderbartPanel__heading"
            >
              NOK 0,00
            </span>
            <span
              className="ekspanderbartPanel__indikator"
            />
          </div>
        </button>
      </div>
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
.inntekter--111111111--2017-11 {
  grid-area: inntekter--111111111--2017-11;
}
",
        }
      }
    />
    <div
      className="item inntekter inntekter--111111111--2017-11"
    >
      <div
        className="ekspanderbartPanel ekspanderbartPanel--lukket"
      >
        <button
          aria-expanded={false}
          className="ekspanderbartPanel__hode"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <div
            className="ekspanderbartPanel__flex-wrapper"
          >
            <span
              className="typo-ingress ekspanderbartPanel__heading"
            >
              NOK 79 879,40
            </span>
            <span
              className="ekspanderbartPanel__indikator"
            />
          </div>
        </button>
      </div>
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
.maaned--2017-12 {
  grid-area: maaned--2017-12;
}
",
        }
      }
    />
    <div
      className="item maaned maaned--2017-12"
    >
      desember 2017
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
.inntekter--222222222--2017-12 {
  grid-area: inntekter--222222222--2017-12;
}
",
        }
      }
    />
    <div
      className="item inntekter inntekter--222222222--2017-12"
    >
      <div
        className="ekspanderbartPanel ekspanderbartPanel--lukket"
      >
        <button
          aria-expanded={false}
          className="ekspanderbartPanel__hode"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <div
            className="ekspanderbartPanel__flex-wrapper"
          >
            <span
              className="typo-ingress ekspanderbartPanel__heading"
            >
              NOK 48 569,00
            </span>
            <span
              className="ekspanderbartPanel__indikator"
            />
          </div>
        </button>
      </div>
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
.inntekter--111111111--2017-12 {
  grid-area: inntekter--111111111--2017-12;
}
",
        }
      }
    />
    <div
      className="item inntekter inntekter--111111111--2017-12"
    >
      <div
        className="ekspanderbartPanel ekspanderbartPanel--lukket"
      >
        <button
          aria-expanded={false}
          className="ekspanderbartPanel__hode"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <div
            className="ekspanderbartPanel__flex-wrapper"
          >
            <span
              className="typo-ingress ekspanderbartPanel__heading"
            >
              NOK 45 569,00
            </span>
            <span
              className="ekspanderbartPanel__indikator"
            />
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    className="toolbar flex knapprad"
  >
    <div
      className="leggtilarbeidsgiver"
    >
      <button
        className="knapp knapp--standard knapp--mini"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Legg til arbeidsgiver
      </button>
    </div>
    <div
      className="flexend flex"
    >
      <div
        className="w200 marginhoyre16"
      >
        <p
          className="typo-undertekst"
        >
          Du må bekrefte at de nye opplysningene skal benyttes
        </p>
      </div>
      <button
        className="knapp knapp--hoved knapp--disabled"
        disabled={true}
        onClick={[Function]}
        type="submit"
      >
        Bekreft
      </button>
    </div>
  </div>
</form>
`;
